src/score_hv/harvesters/global_bucket_precip_ave.py

```diff
@@ -0,0 +1,129 @@
+VALID_VARIABLES  = ('prateb_ave')
+
+HarvestedData = namedtuple('HarvestedData', ['filenames',
+                                             'statistic',
+                                             'variable',
+                                             'value',
+                                             'units'])
+@dataclass
+class GlobalBucketPrecipRateConfig(ConfigInterface):
+
+    config_data: dict = field(default_factory=dict)
+
+    def __post_init__(self):
+        self.set_config()
+
+    def set_config(self):
+        """ function to set configuration variables from given dictionary
+        """
+        self.harvest_filenames = self.config_data.get('filenames')
+        self.set_stats()
+        self.set_variables()
+
+    def set_variables(self):
+        """
+        set the variables specified by the config dict
+        """
+        
+        self.variables = self.config_data.get('variable')
+        for var in self.variables:
+            if var not in VALID_VARIABLES:
+                msg = ("'%s' is not a supported global bucket precip average "
+                       "variable to harvest from the background forecast data. "
+                       "Please reconfigure the input dictionary using only the "
+                       "following variables: %r" % (var, VALID_VARIABLES))
+                raise KeyError(msg)
+    
+    def get_stats(self):
+        ''' return list of all stat types based on harvest_config '''
+        return self.stats
+
+    def set_stats(self):
+        """
+        set the statistics specified by the config dict
+        """
+        self.stats = self.config_data.get('statistic')
+        for stat in self.stats:
+            if stat not in VALID_STATISTICS:
+                msg = ("'%s' is not a supported statistic to harvest for global bucket precipitation. "
+                       "Please reconfigure the input dictionary using only the "
+                       "following statistics: %r" % (stat, VALID_STATISTICS))
+                raise KeyError(msg)
+    
+
+
+    def get_variables(self):
+        ''' return list of all variable types based on harvest_config'''
+        return self.variables
+
+@dataclass
+
+
+@dataclass
+class GlobalBucketPrecipRateHv(object):
+    """ Harvester dataclass used to parse precip stored in 
+        background forecast data
+    
+        Parameters:
+        -----------
+        config: TemperatureConfig object containing information used to determine
+                which variable to parse the log file
+        Methods:
+        --------
+        get_data: parse descriptive statistics from log files based on input
+                  config file
+    
+                  returns a list of tuples containing specific data
+    """
+    config: GlobalBucketPrecipRateConfig = field(
+                                default_factory=GlobalBucketPrecipRateConfig)
+    
+    def get_data(self):
+        """ Harvests prateb_ave from background forecast data
```

Comment:
recommending to not use a hardcoded variable name prateb_ave in the docstring since it is not a hardcoded value in the code described, would recommend generic terms 'harvests requested statistics and variables'